### PR TITLE
Creating and Joining lobbies

### PR DESCRIPTION
**Pull request by Grizzlytron**
I implemented the code such that joining and creating lobbies is possible as users.  Also added other goodies like the Copybutton. This is still a work in progress but would like to get some feedback and ideas on how to improve. (As a note: I wasn't able to fully rebase this branch onto main beforehands, due to some merging issues with the pictures. I will try that later on again, therefore images aren't up to date as on main page.)

### CODE REVIEW

**Review by na50r — APPROVED** [review]
I tested it out and it worked on my side.
As for the profile pictures, I will work on a way such that they can persist in the database and then it should be doable to also use them in your implementation. So you can either wait for me or merge ahead and I do the changes involving the profile pictures.

File: src/components/views/LobbyOverview.tsx [hunk 1]
```diff
@@ -54,60 +56,109 @@ const LobbyOverview = () => {
...
+            /* For some reason it is necessary to make the post request with
+             * a requestbody, if i leave it out, the post request returns an error. */
+
+            const config = {headers: {userToken: userToken}};
+            const response = await api.post("/lobbies/" + lobbyCode + "/players", requestbody, config);
+            console.log(response);
+            navigate("/lobby/" + lobbyCode)
+        } catch (error) {
+            handleError(error);
+            alert(("Unable to join lobby with lobbycode: " + lobbyCode));
```
Comment (Tmmn) [inline src/components/views/LobbyOverview.tsx:80]: add some more specific error handling or/and display error message that was sent from the server so the user knows what the reason for the failure was and can take action (logging in for example)

File: src/components/views/LobbyOverview.tsx [hunk 2]
```diff
@@ -54,60 +56,109 @@ const LobbyOverview = () => {
...
+            const requestbody = {"anonymous" : false}
+            const config = {headers: {userToken: userToken}}
+            const response = await api.post("/lobbies", requestbody, config);
+            const createdLobby = response.data.lobby;
+            console.log(createdLobby);
+
+            navigate("/lobby/" + createdLobby.code);
+        }     catch (error) {
+            handleError(error)
+            alert("There was en error creating your lobby, check the log for details");
```
Comment (Tmmn) [inline src/components/views/LobbyOverview.tsx:96]: same as in join lobby

File: src/models/Player.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/**
+ * Player model
+ */
+class Player {
+    constructor(data = {}) {
+        this.username = null;
```
Comment (Tmmn) [inline src/models/Player.js:6]: is it not playername

File: src/models/User.js [hunk 4]
```diff
@@ -4,12 +4,13 @@
 class User {
     constructor(data = {}) {
         this.id = null;
-        this.username = null;
+        this.name = null;
```
Comment (Tmmn) [inline src/models/User.js:7]: is it not username?

**Review by Tmmn — COMMENTED** [review]
Everything important except the quit request is working. I added some comments and if you have time and it's easy to do, can you make it that the code gets cleared from the input field when I de-select a lobby? And is it intentional that the start game and quit buttons are animated very differently than every other button that we used so far?

File: src/components/views/Lobby.tsx [hunk 5]
```diff
@@ -34,31 +37,36 @@ const gamemodes= [
...
+                console.log(lobby);
+                console.log(players);
+
+            } catch (error) {
+                handleError(error)
+                alert("Unable to display lobby data");
+            }
+        };
+        fetchLobbyAndPlayers();
+    }, [selectedGamemode]); //By adding this dependency, everytime a gamemode is selected, the players get updated
```
[the remark this answers is not in the document]
Comment (Grizzlytron) [reply src/components/views/Lobby.tsx:69]: Okay, makes sense. Websockets seem like the way to go.

File: src/models/User.js (hunk 4 above)
Comment (Grizzlytron) [reply src/models/User.js:7]: good catch

File: src/models/Player.js (hunk 3 above)
Comment (Grizzlytron) [reply src/models/Player.js:6]: yes. good one

**Comment by Tmmn** [comment]
TODO: button to switch visibility public/private, field to change lobby name and disable game mode buttons for non owner players.

**Review by Tmmn — APPROVED** [review]

**Pull request merged**